packages/lime-system/files/usr/lib/lua/lime/network.lua

```diff
@@ -153,6 +153,8 @@ function network.setup_dns()
 			uci:set("dhcp", s[".name"], "local", "/"..cloudDomain.."/")
 			uci:set("dhcp", s[".name"], "expandhosts", "1")
 			uci:set("dhcp", s[".name"], "domainneeded", "1")
+			--! allow queries from non-local ips (i.e. from other clouds)
+			uci:set("dhcp", s[".name"], "localservice", "0")
```

Comment:
Does this mean that our dnsmasq ill attempt to answhere any type of queries or only the ones relatives to the domain of the cloud?  The latter is ok, the first could expose our dnsmasq to become a DNS botnet considering a community network that have public IPv6

Reply:
any type of queries (including recursive), as far as i understand.
as long as `fw3` is installed and running (which is, by default), it will block any incoming queries over the public internet, both on ipv4 and ipv6. maybe `lime-firewall` does this as well by default? (it really should)
ideally, instead of disabling `local-service`, we would enable `auth-server` in dnsmasq conf. but the standard `dnsmasq` package shipped with openwrt does not compile in `auth` support. (`dnsmasq-full` does, with a myriad other features)

Reply:
How does `fw3` distinguish from a public IPv6 from the internet and a public IPv6 from another cloud? Maybe we should create a package `dnsmasq-authoritative` package with just the options we need, and eventually upstream it?

Reply:
`fw3` AFAIU distinguish from incoming interface. if it's WAN port, it blocks incoming traffic.

creating `dnsmasq-auth` sounds great in principle... in my experience, it's easy to somehow during menuconfig end up with dnsmasq (vanilla) installed. and quite frustrating to debug why `auth-server` is not working

Reply:
we had this issue with `dnsmasq-dhcpv6` (instead of `dnsmasq`)

Reply:
Apparently that have been solved and now if the wrong dnsmasq is selected it fails at compile time

Reply:
> `fw3` AFAIU distinguish from incoming interface. if it's WAN port, it blocks incoming traffic.

Well that's not good as we support WAN-WAN connections as a way to connect different clouds, thought I think we can eventually change that, with the current config system that would not be that problematic IMHO

Reply:
> as we support WAN-WAN connections as a way to connect different clouds

We never did, as far as I know the right way to connect different clouds is a lan-lan connection where the two clouds have different batman-adv vlan id.
People where using wan-wan because batman-adv was not running there, I suppose, but with the new release batman-adv runs also on wan so using wan-wan connection makes even less sense than before.

Reply:
> as long as fw3 is installed and running (which is, by default)

fw3 is not installed by default, see the instructions in the [development page](https://libremesh.org/development.html).
For more discussion about whether selecting fw3 or not see #280

Reply:
> We never did, as far as I know the right way to connect different clouds is a lan-lan connection where the two clouds have different batman-adv vlan id.
> People where using wan-wan because batman-adv was not running there, I suppose, but with the new release batman-adv runs also on wan so using wan-wan connection makes even less sense than before.

Nope you are missing a piece about this, if you connect two cloud LAN-LAN it will become a huge mess, all the MAC addresses of the second cloud would be seen as batman-adv clients in the second cloud, telling batman-adv to send hello packets on LAN doesn't help in this case even if it helps a lot in other.

Reply:
> > `fw3` AFAIU distinguish from incoming interface. if it's WAN port, it blocks incoming traffic.
> 
> Well that's not good as we support WAN-WAN connections as a way to connect different clouds, thought I think we can eventually change that, with the current config system that would not be that problematic IMHO

Also how does `fw3` distinguish between a request coming to resolve something like `myraspberry.mycommunity.info` which is legitimate even from WAN and one that requests `www.mybllaaaaaaaadomain.com`. Using `fw3` or any other firewall to do this will break that legitimate use case which is one of the purposes we are implementing this.

Reply:
> > We never did, as far as I know the right way to connect different clouds is a lan-lan connection where the two clouds have different batman-adv vlan id.
> > People where using wan-wan because batman-adv was not running there, I suppose, but with the new release batman-adv runs also on wan so using wan-wan connection makes even less sense than before.
> 
> Nope you are missing a piece about this, if you connect two cloud LAN-LAN it will become a huge mess, all the MAC addresses of the second cloud would be seen as batman-adv clients in the second cloud, telling batman-adv to send hello packets on LAN doesn't help in this case even if it helps a lot in other.

Sorry but I still don't understand.
From my point of view, the fact that Batman-adv runs on VLAN both on LAN than on WAN makes them identical for Baman-adv.
And I repeat: Batman-adv is also running on WAN (with VLAN, identical as on LAN).